### ISSUE TITLE
Continuing setting up the initial page causes exception

### ISSUE THREAD

**Issue by Luo-Liang** [OrchardCMS/OrchardCore#302]
I'm running with .net core on Visual Studio. I'm starting up Orchard.CMS.Web project.
When I get pass the setup page, an error in HomepageRoute.GetVirtualPath thrown. The reason is _token is null in :
foreach (var entry in _tokens)
I'm not sure why the _token is null.
Now restart the site, and try to continuing setting up again. There will be a different exception thrown. The UpdateShellDescriptorAsync code in ShellDescriptorManager throws.
        if (priorSerialNumber != serialNumber)
        {
            throw new InvalidOperationException("Invalid serial number for shell descriptor");
        }
This can be resolved by deleting the database tables, but then it throws the NullReferenceException in the first step.
When this happens, the httpContext points to home/display/{number}, where {number} increments each time I retry.
The following stacktrace captures this.All these can be 100% reproduced on my machine.
Kindly let me know if there is obvious mistake running it.

<img width="1016" alt="4100177a-9ef1-11e6-8c8a-3e7fbc801806" src="https://cloud.githubusercontent.com/assets/1490925/19869081/028e0156-9f68-11e6-94ce-930d33d5d70f.png">

**Comment by PinpointTownes** [comment]
@sebastienros FYI, I fixed it in the `oauth2` branch a while ago (I was blocked by this bug too): https://github.com/OrchardCMS/Orchard2/commit/90e17c3b5e60ee90d471327dda1ab8c0bbe75229.

**Comment by sebastienros** [comment]
I see, I have also fixed it, on the tenants branch though, and differently, it's a race condition.

**Comment by sebastienros** [comment]
As soon as the tenants branch is merged, this will be fixed, tomorrow probably.

**Comment by PinpointTownes** [comment]
@sebastienros looks like the `tenants` branch was merged into `master`. Any chance you could merge `master` into `oauth2` and undo my workaround? Thanks :smile:

**Comment by sebastienros** [comment]
I can rebase your branch on master if you both agree, but if you are working on you might lose some stuff. Either both of you tell me to do it or one of you has to do it.

**Comment by jersiovic** [comment]
For me there is no problem on doing a rebase, I have no work in progress. If you want I can do the rebase now
